src/settings/LoanHistory/ExceptionsList.test.js

```diff
@@ -0,0 +1,146 @@
+
+jest.mock('./ExceptionCard', () => jest.fn(({
+  onRemoveException,
+  exceptionIndex,
+}) => {
+  return (
+    // eslint-disable-next-line
+    <div
+      data-testid={`exeptionCard-${exceptionIndex + 1}`}
+      onClick={() => onRemoveException(exceptionIndex)}
+    />
+  );
+}));
+
+describe('ExceptionsList', () => {
+  const mockedFields = {
+    map: jest.fn((callback) => mockedFields.values.map(callback)),
+    push: jest.fn(),
+    remove: jest.fn(),
+    values: [
+      'firstTestPatch',
+      'secondTestPatch',
+    ],
+  };
+  const mockedResources = {
+    paymentMethods: {
+      records: [
+        {
+          payments: [
+            {
+              nameMethod: 'money',
+            },
+            {
+              nameMethod: 'card',
+            },
+            {
+              nameMethod: 'card',
+            },
+          ],
+        },
+      ],
+    },
+  };
+  const testIds = {
+    header: 'header',
+    exceptions: 'exceptionSection',
+    button: 'addButton',
+  };
+  const labelIds = {
+    header: 'ui-circulation.settings.loanHistory.exceptionSectionHeader',
+    button: 'ui-circulation.settings.loanHistory.addExceptionButton',
+  };
+
+  const getById = (id) => within(screen.getByTestId(id));
+  const exeptionCardTest = (patch, index) => {
+    const number = index + 1;
+    const expectedResult = {
+      pathToException: patch,
+      paymentMethods: [{
+        label: 'money',
+        value: 'money',
+      },
+      {
+        label: 'card',
+        value: 'card',
+      },
+      ],
+      exceptionIndex: index,
+    };
+
+    it(`should execute ${number} "ExceptionCard" with passed props`, () => {
+      expect(ExceptionCard).toHaveBeenNthCalledWith(number, expect.objectContaining(expectedResult), {});
+    });
+
+    it(`should correctly execute "handleRemoveField" for ${number} "ExceptionCard"`, () => {
+      expect(mockedFields.remove).toHaveBeenCalledTimes(0);
+      fireEvent.click(screen.getByTestId(`exeptionCard-${number}`));
+      expect(mockedFields.remove).toHaveBeenCalledWith(index);
+    });
+  };
+
+  beforeEach(() => {
+    render(
+      <ExceptionsList
+        fields={mockedFields}
+        resources={mockedResources}
+      />
+    );
+  });
+
+  afterEach(() => {
+    ExceptionCard.mockClear();
+    mockedFields.push.mockClear();
+    mockedFields.remove.mockClear();
+  });
+
+  it('component should have correct manifest', () => {
+    const expectedResult = {
+      paymentMethods: {
+        type: 'okapi',
+        path: 'payments',
+        throwsErrors: false,
+      },
+    };
+
+    expect(ExceptionsList.manifest).toEqual(expectedResult);
+  });
+
+  it('should render header label', () => {
+    expect(getById(testIds.header).getByText(labelIds.header)).toBeVisible();
+  });
+
+  it('"exceptions" section should have correct class', () => {
+    expect(screen.getByTestId(testIds.exceptions)).toHaveAttribute('class', css.exceptionSection);
```

Comment:
@Dmitriy-Litvinenko should we test css classes? On the one hand it is useful. On the other hand we test styling but not logic.

Reply:
On my opinion, we do not need tests that check for the presence of css class.
@Stanislau-Snapok Could you please remove them?
CC: @pavelspichonak , @Stanislau-Snapok

Reply:
deleted